jt-bandmerge/src/test/java/it/geosolutions/jaiext/bandmerge/BandMergeTest.java

```diff
@@ -656,4 +662,92 @@ private void testExtendedBandMerge(RenderedImage[] sources, boolean noDataUsed,
+        testBandBergeOnBandSelected(DataBuffer.TYPE_FLOAT, (float) 0, (float) 1);
+        testBandBergeOnBandSelected(DataBuffer.TYPE_DOUBLE, (double) 0, (double) 1);
+    }
+    
+    public void testBandBergeOnBandSelected(int dataType, Number noDataValue, Number dataValue) {
+        testBandBergeOnBandSelected(dataType, noDataValue, dataValue, true);
+        testBandBergeOnBandSelected(dataType, noDataValue, dataValue, false);
+        testBandBergeOnBandSelected(dataType, null, dataValue, true);
+        testBandBergeOnBandSelected(dataType, null, dataValue, false);
+    }
+
+    private void testBandBergeOnBandSelected(int dataType, Number noDataValue, Number dataValue,
+            boolean addROI) {
+        Range[] nodata;
+        if(noDataValue == null) {
+            nodata = null;
+        } else {
+            switch (dataType) {
+            case DataBuffer.TYPE_BYTE: {
+                final byte noDataPrimitive = noDataValue.byteValue();
+                nodata = new Range[] { RangeFactory.create(noDataPrimitive, noDataPrimitive) };
+                break;
+            }
+            case DataBuffer.TYPE_SHORT: {
+                final short noDataPrimitive = noDataValue.shortValue();
+                nodata = new Range[] { RangeFactory.create(noDataPrimitive, noDataPrimitive) };
+                break;
+            }
+            case DataBuffer.TYPE_USHORT:
+            case DataBuffer.TYPE_INT: {
+                final int  noDataPrimitive = noDataValue.intValue();
+                nodata = new Range[] { RangeFactory.create(noDataPrimitive, noDataPrimitive) };
+                break;
+            }
+            case DataBuffer.TYPE_FLOAT: {
+                final float noDataPrimitive = noDataValue.floatValue();
+                nodata = new Range[] { RangeFactory.create(noDataPrimitive, noDataPrimitive) };
+                break;
+            }
+            case DataBuffer.TYPE_DOUBLE: {
+                final double noDataPrimitive = noDataValue.doubleValue();
+                nodata = new Range[] { RangeFactory.create(noDataPrimitive, noDataPrimitive) };
+                break;
+            }
+    
+            default:
+                throw new IllegalArgumentException();
+            }
+        }
+        
+        TiledImage image = (TiledImage) createTestImage(dataType, 10, 10, noDataValue, false, 3,
+                dataValue);
+        for (int i = 0; i < image.getHeight(); i++) {
+            for (int j = 0; j < image.getWidth(); j++) {
+                for (int b = 0; b < image.getNumBands(); b++) {
+                    int value = b == image.getNumBands() - 1 ? 1 : 0;
+                    image.setSample(j, i, b, value);
+                }
+            }
+        }
+        
+        RenderedOp selected = BandSelectDescriptor.create(image, new int[] {2}, null);
+        
+        RenderedOp bandMerged;
+        if(addROI) {
+            ROI roi = new ROIShape(new Rectangle2D.Double(0, 0, 100, 100));
```

Comment:
Oh, just because building it is less code than ROIGeometry, and this is just a test :-p